File: FibonacciSeries/Ruby/fib.rb
# fib series solution in ruby for code eval by steven a dunn

def fib(n)
  if n == 0 then return 0 end
  if n == 1 then return 1 end

  f1 = 0
  f2 = 1

  2.upto(n) do 
    tmp = f2
    f2 += f1
    f1 = tmp
  end

  return f2
end

File.open(ARGV[0]).each_line do |line|
  n = line.to_i
  puts fib(n)
end

File: FibonacciSeries/JS/fib.js
// Fib seq soln in Node for CodeEval.com by Steven A. Dunn

var fs = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
  if (line !== "") {
    var n = parseInt(line);
    var buf1 = 1;
    var buf2 = 1;
    var temp = 0;

    if (n === 0) console.log(0);
    else if (n == 1 || n == 2) console.log(1);
    else {
      for (var fibIdx = 3; fibIdx <= n; ++fibIdx) {
        temp = buf1;
        buf1 = buf2;
        buf2 = buf1 + temp;
      }  
      console.log(buf2);
    }
  }
});
